frappe/public/js/frappe/form/controls/link.js

```diff
@@ -69,6 +71,84 @@ frappe.ui.form.ControlLink = class ControlLink extends frappe.ui.form.ControlDat
 			this.$input_area.find(".link-btn").remove();
 		}
 	}
+	set_formatted_input(value) {
+		super.set_formatted_input();
+		if (!value) return;
+		let doctype = this.get_options();
+		this.set_data_value(frappe.get_link_title(doctype, value) || value, value);
+	}
+	set_data_value(link_display, value) {
+		if (!this.$input) {
+			return;
+		}
+
+		this.$input.val(link_display);
+		this.data_value = value;
+	}
+	parse_validate_and_set_in_model(value, label, e) {
+		if (this.parse) value = this.parse(value, label);
+		if (label) {
+			this.label = label;
+			frappe.add_link_title(this.df.options, value, label);
+		}
+
+		return this.validate_and_set_in_model(value, e);
+	}
+	validate_and_set_in_model(value, e) {
+		var me = this;
+		let force_value_set = (this.doc && this.doc.__run_link_triggers);
+		let is_value_same = (this.get_model_value() === value);
+
+		if (this.inside_change_event || (!force_value_set && is_value_same)) {
+			return Promise.resolve();
+		}
+
+		this.inside_change_event = true;
+		function set(value) {
+			me.inside_change_event = false;
+			return frappe.run_serially([
+				() => me._validated = true,
+				() => me.set_model_value(value),
+				() => delete me._validated,
+				() => {
+					me.set_mandatory && me.set_mandatory(value);
+
+					if(me.df.change || me.df.onchange) {
+						// onchange event specified in df
+						frappe.set_link_title(me);
+						let set = (me.df.change || me.df.onchange).apply(me, [e]);
+						me.set_invalid && me.set_invalid();
+						return set;
+					}
+					me.set_invalid && me.set_invalid();
+				}
+			]);
+		};
+		value = this.validate(value);
+		if (value && value.then) {
+			// got a promise
+			return value.then((value) => set(value));
+		} else {
+			// all clear
+			return set(value);
+		}
+	}
```

Comment:
Is there a way to avoid overwriting this method? This is done only to add one line change `frappe.set_link_title(me);` before calling `onchange` method.